### PR TITLE
unk_0200ACF0

### PR DESCRIPTION
**Pull request by PikalaxALT**
follows from #300

### CODE REVIEW

File: src/unk_0200ACF0.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+#include "unk_0200ACF0.h"
+#include "unk_020215A0.h"
+#include "unk_02022588.h"
+
+BOOL sub_0200ACF0(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(obj) == GF_GFX_RES_TYPE_CHAR);
+
+    UnkStruct_02021758 sp00;
+    sp00.charData = GF2DGfxResObj_GetCharDataPtr(obj);
```
Comment (adrienntindall) [inline src/unk_0200ACF0.c:11]: sp notation

Comment (adrienntindall) [reply src/unk_0200ACF0.c:11]: etc for file

File: src/unk_0200ACF0.c [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        }
+    }
+}
+
+NNSG2dImageProxy *sub_0200AF00(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(obj) == GF_GFX_RES_TYPE_CHAR);
+
+    int id = GF2DGfxResObj_GetResID(obj);
+    return sub_02021910(id);
```
Comment (adrienntindall) [inline src/unk_0200ACF0.c:111]: sub_02021910(GF2DGfxResObj_GetResID(obj))

File: src/unk_0200ACF0.c [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    }
+    return ret;
+}
+
+void sub_0200AF80(const NNSG2dImageProxy* imgProxy) {
+    GF_ASSERT(imgProxy != NULL);
+    sub_02021A50(imgProxy);
+}
+
+// ------------------
```
Comment (adrienntindall) [inline src/unk_0200ACF0.c:136]: ?

**Review by adrienntindall — CHANGES_REQUESTED** [review]
Check also comments on the other PR, I mightve missed something for this fil

File: src/unk_0200ACF0.c (hunk 1 above)
Comment (PikalaxALT) [reply src/unk_0200ACF0.c:11]: keeping it in place for objects of type (pointer to) UnkStruct_XXXXXXXX

File: src/unk_0200ACF0.c (hunk 3 above)
Comment (PikalaxALT) [reply src/unk_0200ACF0.c:136]: probably marking a border between two logical (to me at the time) groups of functions without a good-enough explanation of what that logical difference was

File: src/unk_0200ACF0.c [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    int resID = GF2DGfxResObj_GetResID(charResObj);
+    NNSG2dCellDataBank *cellData = GF2DGfxResObj_GetCellDataPtr(cellResObj);
+    ret = sub_02021934(resID, cellData->pVramTransferData->szByteMax);
+    if (ret == NULL) {
+        ret = sub_020219B8(sub_02021910(resID));
+    }
+    return ret;
+}
+
+void sub_0200AF80(const NNSG2dImageProxy* imgProxy) {
```
Comment (red031000) [inline src/unk_0200ACF0.c:129]: pointer alignment

File: src/unk_0200ACF0.c [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+void sub_0200B050(GF_2DGfxResObjList *plttResObjList) {
+    GF_ASSERT(plttResObjList != NULL);
+    for (int i = 0; i < plttResObjList->max; ++i) {
+        if (plttResObjList->obj[i] != NULL) {
+            GF_ASSERT(sub_0200B00C(plttResObjList->obj[i]));
+        }
+    }
+}
+
+void sub_0200B084(GF_2DGfxResObj* plttResObj) {
```
Comment (red031000) [inline src/unk_0200ACF0.c:176]: pointer alignment

File: src/unk_0200ACF0.c [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+    int resID = GF2DGfxResObj_GetResID(plttResObj);
+    if (imageProxy != NULL) {
+        return sub_020227D4(resID, imageProxy);
+    } else {
+        return sub_020227AC(resID);
+    }
+}
+
+int sub_0200B12C(GF_2DGfxResObj* plttResObj, NNS_G2D_VRAM_TYPE vram) {
```
Comment (red031000) [inline src/unk_0200ACF0.c:211]: pointer alignment

**Review by red031000 — CHANGES_REQUESTED** [review]
small things

**Review by red031000 — APPROVED** [review]
I'd rather s32 is used than int, but I know matching can get weird with it so it's fine

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**